src/book/02-system/01-nubots/09-config-script.mdx

```diff
@@ -0,0 +1,65 @@
+---
+section: System
+chapter: NUbots
+title: Configuration and Script System
+description: Introduction to then configuration and script system
+slug: /system/nubots/config-script
+---
+
+## The Configuration System
+
+The NUbots configuration system is based on a hierarchical [YAML](https://yaml.org/) file system.
+
+The hierarchical structure is as follows
+
+```text
+config/
+├── nugus1/
+│   └── ExampleModule.yaml
+├── nugus2/
+│   └── ExampleModule.yaml
+├── robocup/
+│   └── ExampleModule.yaml
+└── ExampleModule.yaml
+```
+
+The file `config/ExampleModule.yaml` must exist for every module. This file defines all of the configuration for the module.
+`config/nugus1/ExampleModule.yaml` is an optional file and allows you to provide robot-specific overrides for specific configuration values, in this case it would provide overrides for the `nugus1` robot.
+Similarly, `config/robocup/ExampleModule.yaml` allows for the `robocup` role to further override configuration values.
+
+It is intended that all configuration values exist in the default configuration file and that only a subset of configuration values appear in the robot-specific and role-specific override files.
```

Comment:
Doesn't always happen, for example the cameras don't have a global `serial_number` field since they don't make sense in a general sense, but are required in the per robot settings.